### DISCUSSION TITLE
How do I run commands concurrently?

### DISCUSSION THREAD

**Discussion by prakharrr** [community/community#26864] [Actions]
<p>I have a project uses interprocess-communication using RabbitMQ.</p>
<p>I want to run rabbitmq in one shell and consumer and publisher in separate instances.</p>
<p>For example:</p>
<ol>
<li>
<p>rabbitmq-server</p>
</li>
<li>
<p>While that rabbitmq-server is running, I want to run a python service <code>_python3 receiver.py_</code> in another separate shell</p>
</li>
<li>
<p>In another instance I want to be able to run another service <code>_python3 sender.py_</code> and <code>_python3 passive\_sender.py_</code></p>
</li>
</ol>
<p>Sorry I’m new to creating pipelines and workflows.</p>

**Comment by ethomson** [comment, marked as answer]
<p>It sounds like you might want to run the server in the background?  The <code>&amp;</code> symbol at the end of a command in bash will run the command in the background.  For example, to run rabbitmq-server in the background, and the receiver _also_ in the background, and then run (blocking) on the sender:</p>
<ul>
<li>run: |</li>
</ul>
<p>    rabbitmq-server &amp;</p>
<p>    python3 receiver.py &amp;</p>
<p>    python3 sender.py</p>

**Comment by gengjiawen** [comment]
<p>This doesn’t seems work on windows. The script won’t report error, but looks like not execute in the end. My case is using <code>choco install nasm vscode &amp; </code></p>
